components/brave_wallet_ui/image/image.js

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+document.addEventListener('DOMContentLoaded', function () {
+  const urlParam = new URLSearchParams(window.location.search).get('targetUrl')
+  const imgUrl = urlParam ? urlParam.replace('chrome://image?', '') || '' : ''
+
+  const imageElement = document.getElementById('untrusted-image') // as HTMLImageElement | null
+
+  if (imageElement && imgUrl) {
```

Comment:
please check here that `imgUrl.startsWith('https://assets.coingecko.com/')`

or if that's not doable then at least check that it starts with https://